### ISSUE TITLE
address view does not show unconfirmed transactions

### ISSUE THREAD

**Issue by raedah** [dcrdata/dcrdata#277]

**Comment by papacarp** [comment]
Some sleuthing guidance for this one:
1) if an address has no transactions in it previously, then addressHistory fails and it falls back to RPC to display the transactions which includes unconfirmed transactions.

https://github.com/decred/dcrdata/blob/e22ff164f2e24e11c2b2fecd712821d008db7a43/explorer/explorerroutes.go#L353-L358

Thus we are able to see unconfirmed transactions sometimes.
![image](https://user-images.githubusercontent.com/11194546/40158909-c16810ea-595b-11e8-9444-d8c6c76c6d7c.png)

2) if an address *does* have transactions in it previously, then AddressHistory succeeds, unconfirmed transactions are retrieved, but never merged into the resulting output.  

https://github.com/decred/dcrdata/blob/e22ff164f2e24e11c2b2fecd712821d008db7a43/explorer/explorerroutes.go#L441

They are just left in a the var uctxn and ignored.  The fix is to merge these into the addrData.Transactions list, and likely re-slice that list to adhere to the start/limit inputs.

_This is my theory as I haven't noticed an unconfirmed transaction go through that has previous history in it to verify and my testnet is not up right now._

If this is not  urgent then due to the rework happening on addresses table this should likely be fixed after we sort out the disposition of #427 by either @dmigwi or I.  If urgent I can put a PR in for it based on existing architecture.

**Comment by chappjc** [comment]
Yes, this is absolutely right.  I had started a branch fixing these issues, but never finished up.  https://github.com/chappjc/dcrdata/commits/addr-page-fixes

I must have stashed teh changes related to the above issues though.

**Comment by chappjc** [comment]
Please feel free to pick this up of course.  But do not wait on PR #426 .  It's big and breaking, so we're not going to rush it through.  Although it's getting closer.

**Comment by dmigwi** [comment]
Since the unconfirmed transactions are already available in the frontend and the only problem is when they are displayed why not add a link to the UNCONFIRMED metric value such that when the link is clicked the unconfirmed txs appear. 

![hyperlink](https://user-images.githubusercontent.com/22055953/40200494-1dbd8be4-5a25-11e8-8f44-bf9f4554db13.png)

I suggest we add the link to the box in red. Whats your take on this @papacarp?

**Comment by chappjc** [comment]
Correction: I meant do not wait on *PR #427 *

**Comment by raedah** [comment]
I would do it the way @papacarp suggested, "merge these into the addrData.Transactions list, and likely re-slice that list to adhere to the start/limit inputs." That way the information is inline and does not require additional user interaction to access.

**Comment by chappjc** [comment]
I agree with @raedah

**Comment by papacarp** [comment]
Ok.  I will put in a PR for it.
